shark/src/main/java/shark/internal/PathFinder.kt

```diff
@@ -382,7 +385,7 @@ internal class PathFinder(
         continue
       }
 
-      val objectId = staticField.value.asObjectId!!
+      val objectId = (staticField.value.holder as ReferenceHolder).value
```

Comment:
As an alternative I could've added `staticField.value.asObjectIdOrThrow` method, but that would be changing the public API. 
We were using `asObjectId!!` which means that we are sure that the holder inside value is a `ReferenceHolder`. This implies that we know what's going on and are ok with replacing this assumption on equivalent of `staticField.value.holder as ReferenceHolder`.

Decompiled Java.
With optimization:
```
ValueHolder var12 = staticField.getValue().getHolder();
if (var12 == null) {
   throw new TypeCastException("null cannot be cast to non-null type shark.ValueHolder.ReferenceHolder");
}
long objectId = ((ReferenceHolder)var12).getValue();
```

Without optimization:
```
Long var12 = staticField.getValue().getAsObjectId();
if (var12 == null) {
   Intrinsics.throwNpe();
}
long objectId = var12;
```

Reply:
Add comment?